### PR TITLE
issues #3845 and #3885 - update release process for LinuxForHealth

### PR DESCRIPTION
**Pull request by lmsurpre**
1. maven artifacts will now be staged at s01.oss.sonatype.org with a groupId of org.linuxforhealth.fhir
2. docker images will now be published to GitHub Packages under the LinuxForHealth organization (ghcr.io/LinuxForHealth)

Note: for the "rebuild" workflow, it checks out the previously tagged version of both the code AND the release automation, so 4.11.0 and 4.11.1 will continue to go to ibmcom/ibm-fhir-server on docker hub

### CODE REVIEW

File: .github/workflows/integration.yml [hunk 1]
```diff
@@ -31,7 +31,7 @@ jobs:
         mvn -B org.apache.maven.plugins:maven-dependency-plugin:3.1.2:go-offline -f fhir-parent -DexcludeReactor=true -Dmaven.wagon.http.retryHandler.count=3
         mvn -B org.apache.maven.plugins:maven-dependency-plugin:3.1.2:resolve-plugins -f fhir-parent -DexcludeReactor=true -Dmaven.wagon.http.retryHandler.count=3
         mvn -B install --file fhir-parent -DskipTests -P include-fhir-igs,integration --no-transfer-progress -Dmaven.wagon.http.retryHandler.count=3
-        mvn -B dockerfile:build -f fhir-install --no-transfer-progress -Dmaven.wagon.http.retryHandler.count=3 -Ddocker.buildArg.VERBOSE=false
+        docker build fhir-install --build-arg VERBOSE=false -t linuxforhealth/fhir-server
```
Comment (dixonwhitmire) [inline .github/workflows/integration.yml:34]: Is the docker build process migrating from maven to the docker CLI? I didn't know if there were relevant portions of the POM(s) which could be removed. I did see this [snippet](https://github.com/LinuxForHealth/FHIR/pull/3886/files#diff-0b51d62d390dc0b76f0597ee8788fbb662104fbad227ba8aa5bd43d494fbe3f0L134) but of course I'm not familiar with the overall context, pun intended :)

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -29,14 +29,18 @@ jobs:
         distribution: 'temurin'
         java-version: ${{ matrix.java }}
         cache: 'maven'
+    - name: Log in to GitHub Container Registry
```
Comment (dixonwhitmire) [inline .github/workflows/release.yml:32]: These configurations align with my current understanding of the release workflow. We will also need to ensure we have these credentials backed up in a "secondary" vault, which we can chat about offline.

File: .github/workflows/integration.yml (hunk 1 above)
Comment (lmsurpre) [reply .github/workflows/integration.yml:34]: It's a step in that direction.  Today we have pretty even a mix of `docker build` and `mvn dockerfile:build` in our various workflows.  After this round of changes, we'll use the `docker build` flavor almost exclusively and we'll build to a local image named `linuxforhealth/fhir-server`.

The one place I left the maven-based docker build is in our release workflow.  In that one case we build to `ghcr.io/linuxforhealth/fhir-server`.  See https://github.com/LinuxForHealth/FHIR/blob/c28cb62b3b3188dad33f3086033c8bf5fa279048/build/release/bin/30_release/1_docker.sh#L29-L32

But we don't really *need* to use the dockerfile-maven-plugin there either.  Do you think we should change to use the docker cli there as well?

Additionally we could use the `ghcr.io/linuxforhealth/fhir-server` flavor all over...I just felt that keeping the registry prefix out of the name in the other workflows might help make it more clear that its built locally (and won't fall back to pulling from the registry if we ever fail to build it).

File: .github/workflows/release.yml (hunk 2 above)
Comment (lmsurpre) [reply .github/workflows/release.yml:32]: OK.  I tried to follow your pattern here and I think you had just followed the recommendation of the GitHub docs.  The alternative to using the injected GITHUB_TOKEN is to use a Personal Access Token (PAT).
I'm not sure that either needs to be stored outside of GitHub but I'm willing to hear the argument.

P.S. a related topic for us to chat about is whether we should set "restrictive" permissions for our GITHUB_TOKEN at the org level.  https://docs.github.com/en/actions/security-guides/automatic-token-authentication

**Review by dixonwhitmire — APPROVED** [review]

**Review by PrasannaHegde1 — APPROVED** [review]
LGTM

**Pull request merged**